src/pc/mods/mod.c

```diff
@@ -431,6 +432,11 @@ static void mod_extract_fields(struct Mod* mod) {
             if (snprintf(mod->incompatible, MOD_INCOMPATIBLE_MAX_LENGTH, "%s", extracted) < 0) {
                 LOG_INFO("Truncated mod incompatible field '%s'", mod->incompatible);
             }
+        } else if (mod->category == NULL && (extracted = extract_lua_field("-- category:", buffer))) {
+            mod->category = calloc(MOD_CATEGORY_MAX_LENGTH + 1, sizeof(char));
```

Comment:
you must free this memory when it is not needed, otherwise this could have a memory leak. 
free the memory in `mod_clear` where the other fields are freed, that should be fine.